pkg/model/requests_processing_test.go

```diff
@@ -68,24 +69,71 @@ func testRequestsProcessing(t *testing.T, describe spec.G, it spec.S) {
 			assert.Equal(t, simulator.StockName("RequestsProcessing [99]"), subject.Name())
 		})
 	})
-
 	describe("Add()", func() {
-		var request simulator.Entity
+		var request RequestEntity
 
-		it.Before(func() {
-			bufferStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
-			request = NewRequestEntity(envFake, bufferStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
-			subject.Add(request)
+		describe("there is no free cpu resource", func() {
+			it.Before(func() {
+				*rawSubject.occupiedCPUCapacityMillisPerSecond = *rawSubject.totalCPUCapacityMillisPerSecond
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+			describe("scheduling processing", func() {
+				it("schedules a movement from RequestsProcessing to RequestsFailed", func() {
+					assert.Equal(t, simulator.StockName("RequestsFailed"), envFake.Movements[0].To().Name())
+				})
+			})
+
+			it("there is 1 entity in processingStock", func() {
+				assert.Equal(t, subject.Count(), uint64(1))
+			})
 		})
+		describe("request fails as request total time exceeds request timeout", func() {
+			it.Before(func() {
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 20000, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+
+			it("increments the number of requests since last Stat()", func() {
+				assert.Equal(t, int32(1), rawSubject.numRequestsSinceLast)
+			})
 
-		it("increments the number of requests since last Stat()", func() {
-			assert.Equal(t, int32(1), rawSubject.numRequestsSinceLast)
+			it("cpu resource is allocated", func() {
+				assert.NotEqual(t, rawSubject.occupiedCPUCapacityMillisPerSecond, 0)
+			})
+			it("schedules a movement from RequestsProcessing to RequestsFailed", func() {
+				assert.Equal(t, simulator.StockName("RequestsFailed"), envFake.Movements[0].To().Name())
+			})
+
+			describe("when request timeout is over", func() {
+				it("time for movement from RequestsProcessing to RequestsFailed = current time + timeout ", func() {
+					assert.Equal(t, envFake.TheTime.Add(request.(*requestEntity).requestConfig.Timeout), envFake.Movements[0].OccursAt())
+				})
```

Comment:
This looks like intermediate detail. Would it be possible to test actual results of movements, e.g. that request was counted as failed and that cpuUtiliziation goes down after the timeout (and not before)?

Reply:
Since this is a unit test, not an integration test, it is not supposed to check the result of the Movement execution as part of this test. It is enough to check that the corresponding Movement has been scheduled. As Evironment handles Movements, the result of the Movement execution is out of the scope of the test which is supposed to test requests_processing.

Reply:
Makes sense.